python/pyspark/ml/torch/distributor.py

```diff
@@ -0,0 +1,297 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import math
+from typing import Union, Callable, Optional, Any
+import warnings
+
+from pyspark.sql import SparkSession
+from pyspark.context import SparkContext
+
+
+# TODO(SPARK-41589): will move the functions and tests to an external file
+#       once we are in agreement about which functions should be in utils.py
+def get_conf_boolean(sc: SparkContext, key: str, default_value: str) -> bool:
```

Comment:
Why not just use something like
```
bool(sc.getConf().get(key, default_value))
```
and let Python handle the issue?

Reply:
I believe `bool("false")` would return `True` though since bool doesn't actually attempt to "comprehend" the string value, right? I guess we could do `return sc.getConf().get(key, default_value) == "true"` but that would treat all possible other values here as "false." I guess let's wait for @HyukjinKwon to give his thoughts on the matter.

Reply:
I know some places we ignore invalid configuration values but I believe it's more common to throw an exception on invalid values so I am fine with throwing an exception. I am fine with the current logic.

Reply:
How about using some function like `distutils.util.strtobool` ?

Reply:
Is this [deprecation](https://peps.python.org/pep-0632/) a relevant issue if we were to use `distutils`?

Reply:
In this case, it is fine with me to write a new function. Just consider to make a general utils function for strtobool.

Reply:
@HyukjinKwon Could you add pyspark helper method for handling bool config ? No need to block this PR.